react/ProductSummaryImage.tsx

```diff
@@ -0,0 +1,409 @@
+/* eslint-disable react/prop-types */
+import React, { useState } from 'react'
+import type { PropsWithChildren } from 'react'
+import { CollectionBadges, DiscountBadge } from 'vtex.store-components'
+import classNames from 'classnames'
+import { useDevice } from 'vtex.device-detector'
+import { useResponsiveValues } from 'vtex.responsive-values'
+import type { ResponsiveValuesTypes } from 'vtex.responsive-values'
+import { useCssHandles } from 'vtex.css-handles'
+import type { CssHandlesTypes } from 'vtex.css-handles'
+import { useProduct } from 'vtex.product-context'
+import { ProductSummaryContext } from 'vtex.product-summary-context'
+
+import ImagePlaceholder from './components/ImagePlaceholder'
+import productSummary from './productSummary.css'
+import { changeImageUrlSize } from './utils/normalize'
+import { imageUrl } from './utils/aspectRatioUtil'
+
+const { useProductSummary } = ProductSummaryContext
+
+const CSS_HANDLES = [
+  'image',
+  'imageContainer',
+  'product',
+  'imagePlaceholder',
+] as const
+
+const MAX_SIZE = 500
+const DEFAULT_SIZE = 300
+
+type GetImageSrcParams = {
+  src: string
+  width: number
+  height: number
+  dpi: number
+  aspectRatio?: string
```

Comment:
```suggestion
  aspectRatio?: string | number
```
To match the aspect ratio util